### PR TITLE
feat: use flowevent to replace coherent events

### PR DESCRIPTION
**Pull request by Saschl**
Fixes #[issue_no]

## Summary of Changes

Found an interesting conversation in the WT Discord. 
https://discord.com/channels/750764704175226992/925798646912516117/954353842412195930

Just a prototype, I want to check whether it has advantages for our use cases.

## Screenshots (if necessary)

image

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/shared/src/FlowEventSync.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    public stop() {
+        this.isRunning = false;
+    }
+
+    /**
+     * Processes events received from onInteractionEvent and sends them onto the local bus.
+     * @param target always empty
+     * @param args [0] the bus id [1] SyncDataPackage
+     */
+    processEventsReceived(target, args) {
```
Comment (aguther) [inline src/shared/src/FlowEventSync.ts:55]: target is not being used, consider rename to _target to remove warning

File: src/instruments/src/ND/index.tsx [hunk 2]
```diff
@@ -91,9 +91,11 @@ const NavigationDisplay: React.FC = () => {
 
     const [symbols, setSymbols] = useState<NdSymbol[]>([]);
 
-    useCoherentEvent(`A32NX_EFIS_${side}_SYMBOLS`, (symbols) => {
-        setSymbols(symbols);
-    });
+    useFlowSyncEvent(`A32NX_EFIS_${side}_SYMBOLS`, useCallback((topic, data) => {
```
Comment (aguther) [inline src/instruments/src/ND/index.tsx:94]: topic is not being used, consider rename to _topic to remove warning

File: src/instruments/src/ND/elements/Traffic.tsx [hunk 3]
```diff
@@ -38,9 +38,10 @@ const TCAS_MASK_ROSE: TcasMask = [
 const useAirTraffic = (mapParams, mode) : NdTraffic[] => {
     const [airTraffic, setAirTraffic] = useState<NdTraffic[]>([]);
     const tcasMask = (mode === Mode.ARC ? TCAS_MASK_ARC : TCAS_MASK_ROSE);
-    useCoherentEvent('A32NX_TCAS_TRAFFIC', (airTrafficString: string) => {
-        const newTraffic = JSON.parse(airTrafficString);
-        setAirTraffic(trafficToDisplay(newTraffic, mapParams, tcasMask));
+    useFlowSyncEvent('A32NX_TCAS_TRAFFIC', (topic, data) => {
```
Comment (aguther) [inline src/instruments/src/ND/elements/Traffic.tsx:41]: topic is not being used, consider rename to _topic to remove warning

File: src/instruments/src/ND/elements/FlightPlanVectors.tsx [hunk 4]
```diff
@@ -20,34 +20,12 @@ export interface FlightPlanVectorsProps {
...
-                    if (LnavConfig.DEBUG_PATH_DRAWING) {
-                        console.log(`[ND/Vectors/${EfisVectorsGroup[group]}] Trimmed after end of transmit: oldSize=${old.length} newSize=${ret.length} trimAfter=${trimAfter}`);
-                    }
-
-                    setVectors(ret);
-                }
-
-                return ret;
-            });
+    useFlowSyncEvent(`A32NX_EFIS_VECTORS_${side}_${EfisVectorsGroup[group]}`, useCallback((topic, data) => {
```
Comment (aguther) [inline src/instruments/src/ND/elements/FlightPlanVectors.tsx:26]: topic is not being used, consider rename to _topic to remove warning

**Review by aguther — APPROVED** [review]

File: src/shared/src/FlowEventSync.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
+import stringify from 'safe-stable-stringify';
+
+// TODO replace this with the class from the actual source once the repo is updated
```
Comment (tracernz) [inline src/shared/src/FlowEventSync.ts:3]: What does this TODO refer to?

File: src/shared/src/FlowEventSync.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        const dataObj = stringify(data);
+        // build a data package
+        const dataPackage = {
+            evtNum: this.evtNum++,
+            topic,
+            data: dataObj,
+            isCached,
+        };
+        // queue data package
+        this.dataPackageQueue.push(dataPackage);
```
Comment (tracernz) [inline src/shared/src/FlowEventSync.ts:93]: None of these comments are valuable as they don't add any information that's not obvious from the code.
```suggestion
        const dataObj = stringify(data);

        const dataPackage = {
            evtNum: this.evtNum++,
            topic,
            data: dataObj,
            isCached,
        };

        this.dataPackageQueue.push(dataPackage);
```

**Review by tracernz — COMMENTED** [review]
A suggestion for an even better implementation... a generic event interface with implementations for both flow event and Coherent events.

File: src/shared/src/FlowEventSync.ts (hunk 5 above)
Comment (Saschl) [reply src/shared/src/FlowEventSync.ts:3]: Got most of the code from the NXi package and wanted to replace it with the TS sources once they updated the repo. But not that relevant anymore as I adapted some parts already. WIll remove

**Pull request merged**